lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IdToken } from "../auth/IdToken";
+import { CacheHelpers } from "../cache/CacheHelpers";
+import { ServerAuthorizationTokenResponse } from "../server/ServerAuthorizationTokenResponse";
+import { ScopeSet } from "../auth/ScopeSet";
+import { buildClientInfo } from "../auth/ClientInfo";
+import { Account } from "../auth/Account";
+import { ProtocolUtils } from "../utils/ProtocolUtils";
+import { ICrypto } from "../crypto/ICrypto";
+import { ICacheStorage } from "../cache/ICacheStorage";
+import { TokenResponse } from "./TokenResponse";
+import { PersistentCacheKeys } from "../utils/Constants";
+import { ClientAuthError } from "../error/ClientAuthError";
+
+/**
+ * Class that handles response parsing.
+ */
+export class ResponseHandler {
+    private clientId: string;
+    private cacheStorage: ICacheStorage;
+    private cacheManager: CacheHelpers;
+    private cryptoObj: ICrypto;
+
+    constructor(clientId: string, cacheStorage: ICacheStorage, cacheManager: CacheHelpers, cryptoObj: ICrypto) {
+        this.clientId = clientId;
+        this.cacheStorage = cacheStorage;
+        this.cacheManager = cacheManager;
+        this.cryptoObj = cryptoObj;
+    }
+
+    public setResponseIdToken(originalResponse: TokenResponse, idTokenObj: IdToken) : TokenResponse {
+        if (!originalResponse) {
+            return null;
+        } else if (!idTokenObj) {
+            return originalResponse;
+        }
+    
+        const exp = Number(idTokenObj.claims.exp);
```

Comment:
Can this something like `expiresSeconds`? Make it clear that its the expiration and that it is in seconds.

Reply:
Changed.